backend/src/aggregator.rs

```diff
@@ -1,29 +1,29 @@
-use std::collections::{HashMap, HashSet};
 use actix::prelude::*;
-use actix_web_actors::ws::{CloseReason, CloseCode};
+use actix_web_actors::ws::{CloseCode, CloseReason};
 use ctor::ctor;
+use std::collections::{HashMap, HashSet};
 
-use crate::node::connector::{Mute, NodeConnector};
-use crate::feed::connector::{FeedConnector, Connected, FeedId};
-use crate::util::DenseMap;
+use crate::chain::{self, Chain, ChainId, GetNodeNetworkState, Label};
+use crate::feed::connector::{Connected, FeedConnector, FeedId};
 use crate::feed::{self, FeedMessageSerializer};
-use crate::chain::{self, Chain, ChainId, Label, GetNodeNetworkState};
+use crate::node::connector::{Mute, NodeConnector};
 use crate::types::{ConnId, NodeDetails, NodeId};
+use crate::util::{DenseMap, Hash};
 
 pub struct Aggregator {
+    hashes: HashMap<Hash, ChainId>,
     labels: HashMap<Label, ChainId>,
-    networks: HashMap<Label, ChainId>,
     chains: DenseMap<ChainEntry>,
     feeds: DenseMap<Addr<FeedConnector>>,
     serializer: FeedMessageSerializer,
     /// Denylist for networks we do not want to allow connecting.
-    denylist: HashSet<String>
+    denylist: HashSet<String>,
 }
 
 pub struct ChainEntry {
     addr: Addr<Chain>,
+    hash: Hash,
```

Comment:
Like above, not a fan of the naming here.

Reply:
this is genesis hash too?

Reply:
@niklasad1 Yes.